sqlglot/optimizer/canonicalize.py

```diff
@@ -52,6 +53,17 @@ def remove_redundant_casts(expression: exp.Expression) -> exp.Expression:
     return expression
 
 
+def ensure_bool_predicates(expression: exp.Expression) -> exp.Expression:
+    if isinstance(expression, exp.Connector):
+        _replace_int_predicate(expression.left)
+        _replace_int_predicate(expression.right)
+
+    elif isinstance(expression, (exp.Where, exp.Having)):
```

Comment:
Should we consider `exp.Qualify` as well?

Reply:
I don't know much about QUALIFY. Do we annotate all the window functions properly?

Reply:
Good point, I'm not sure yet. We can tackle this in a future PR then.